File: backend/src/main/java/vn/edu/uit/careerhub/notifications/NotificationService.java
package vn.edu.uit.careerhub.notifications;

import java.util.UUID;

import org.springframework.http.HttpStatus;
import org.springframework.stereotype.Service;

import vn.edu.uit.careerhub.common.AppException;

@Service
public class NotificationService {
    private final NotificationRepository repository;

    public NotificationService(NotificationRepository repository) { this.repository = repository; }

    public NotificationRepository.Result list(UUID userId, int page, int pageSize, boolean unreadOnly) {
        return repository.list(userId, page, pageSize, unreadOnly);
    }
    public long unreadCount(UUID userId) { return repository.unreadCount(userId); }
    public void markRead(UUID userId, UUID notificationId) {
        if (!repository.markRead(userId, notificationId)) {
            throw new AppException(HttpStatus.NOT_FOUND, "NOTIFICATION_NOT_FOUND", "Không tìm thấy thông báo.");
        }
    }
    public void markAllRead(UUID userId) { repository.markAllRead(userId); }
}

File: backend-express-legacy/src/modules/notifications/notification.service.ts
import { AppError } from "../../shared/app-error.js";
import { NotificationRepository } from "./notification.repository.js";
import type { NotificationListQuery } from "./notification.types.js";

export class NotificationService {
  constructor(private readonly repository: NotificationRepository) {}

  list(recipientUserId: string, query: NotificationListQuery) {
    return this.repository.list(recipientUserId, query);
  }

  async unreadCount(recipientUserId: string) {
    return { count: await this.repository.unreadCount(recipientUserId) };
  }

  async markRead(recipientUserId: string, notificationId: string) {
    if (!await this.repository.markRead(recipientUserId, notificationId)) {
      throw new AppError(404, "NOTIFICATION_NOT_FOUND", "Không tìm thấy thông báo.");
    }
  }

  markAllRead(recipientUserId: string) {
    return this.repository.markAllRead(recipientUserId);
  }
}
